### ISSUE TITLE
registerUpload in Linkedin

### ISSUE THREAD

**Issue by asheeshn** [hybridauth/hybridauth#1073]
## Bug, feature or question?
feature

Is this a bug, a feature request or a question?
feature

Give us a short description.
Upload and Use image in linkedin share

### Version and provider
v2 and Linkedin

What version of Hybrid Auth does this relates to?
latest

Is this an issue with a provider? If yes, which one?

### Reproduction

How can we replicate this issue?

**Comment by asheeshn** [comment]
Create an Image Share(3 steps) --- in attachment
Need to post in Linkedin with Image

https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin#create-an-image-share

**Comment by ApacheEx** [comment]
makes sense. Need to decide: should it be extra method which allows uploading image, or something similar like in Twitter provider (https://github.com/hybridauth/hybridauth/blob/master/src/Provider/Twitter.php#L197)

**Comment by ApacheEx** [comment]
I think it makes more sense to post not just image, but also title, link and custom message.

so, the following code should work:

```php
$profile = $linkedin->getUserProfile();

$url = 'https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin';
$image = 'https://docs.microsoft.com/en-us/media/logos/logo-ms-social.png';
$title = 'Check out how to share!';
$message = 'This is custom message.';

$status = [
  'content' => [
    'contentEntities' => [
      [
        'entityLocation' => $url,
        'title' => $title,
        'thumbnails' => [
          [
            'imageSpecificContent' => [
              'width' => 600,
              'height' => 314,
            ],
            'resolvedUrl' => $image,
          ],
        ],
      ],
    ],
    'title' => $title,
  ],
  'distribution' => [
    'linkedInDistributionTarget' => [
      'visibleToGuest' => TRUE,
    ],
  ],
  'owner' => 'urn:li:person:' . $profile->identifier,
  'subject' => $data['title'],
  'text' => [
    'text' => $message,
  ],
];

$linkedin->setUserStatus($status);
```

**Comment by tkaprol** [comment]
> I think it makes more sense to post not just image, but also title, link and custom message.
> 
> so, the following code should work:
> 
> ```
> $profile = $linkedin->getUserProfile();
> 
> $url = 'https://docs.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/share-on-linkedin';
> $image = 'https://docs.microsoft.com/en-us/media/logos/logo-ms-social.png';
> $title = 'Share on LinkedIn - LinkedIn';
> $message = 'This is custom message.';
> 
> $status = [
>   'content' => [
>     'contentEntities' => [
>       [
>         'entityLocation' => $url,
>         'title' => $title,
>         'thumbnails' => [
>           [
>             'imageSpecificContent' => [
>               'width' => 600,
>               'height' => 314,
>             ],
>             'resolvedUrl' => $image,
>           ],
>         ],
>       ],
>     ],
>     'title' => $title,
>   ],
>   'distribution' => [
>     'linkedInDistributionTarget' => [
>       'visibleToGuest' => TRUE,
>     ],
>   ],
>   'owner' => 'urn:li:person:' . $profile->identifier,
>   'subject' => $title,
>   'text' => [
>     'text' => $message,
>   ],
> ];
> 
> $linkedin->setUserStatus($status);
> ```
AFAIK, that doesn't work either. Looks like it is not possible to post to LinkedIn with an image attachment.
